testing/execution.ts

```diff
@@ -1199,6 +1210,70 @@ async function executeSyncFormulaWithContinuations({
   return {result, chainedCommandResults, completion};
 }
 
+async function executeGetPermissionsFormulaWithContinuations({
+  formulaSpecification,
+  params,
+  manifest,
+  executionContext,
+  bundleSourceMapPath,
+  bundlePath,
+  maxRows = DEFAULT_MAX_ROWS,
+  vm,
+}: {
+  formulaSpecification: GetPermissionsFormulaSpecification;
+  params: string[];
+  manifest: BasicPackDefinition;
+  executionContext: SyncExecutionContext;
+  bundleSourceMapPath: string;
+  bundlePath: string;
+  maxRows?: number;
+  vm?: boolean;
+}) {
+  let result = [];
+  let iterations = 1;
+
+  // We need to make a copy of the execution context so we don't pollute the continuation on the Sync formula
+  // if we are running inside of an actual Sync continuation loop.
+  const executionContextCopy: SyncExecutionContext = {
+    ...executionContext,
+    sync: {...executionContext.sync, continuation: undefined},
+  };
+  do {
+    if (iterations > MaxSyncIterations) {
+      throw new Error(
+        `Sync is still running after ${MaxSyncIterations} iterations, this is likely due to an infinite loop.`,
```

Comment:
Nit, would just update the error message so it's clear that the infinite loop was when getting permissions.